source/Trashbot.js

```diff
@@ -16,5 +17,36 @@ export default class Junkership extends Pixi.Sprite {
         if (this.position.x + this.width < 0){
             this.position.x = Reference.GAME_WIDTH
         }
+        this.collisionCheck()
+    }
+
+    collisionCheck(){
+        var x1 = this.position.x
+        var y1 = this.position.y
+        var w1 = this.width
+        var h1 = this.height
+
+        var x2, y2, w2, h2
+
+        game.children.forEach((toCompare) => {
+            if (toCompare instanceof Projectile){
+                x2 = toCompare.position.x
+                y2 = toCompare.position.y
+                w2 = toCompare.width
+                h2 = toCompare.height
+
+                if (x1 < x2 + w2 && x1 + w1 > x2 &&
+                    y1 < y2 + h2 && h1 + y1 > y2) {
+                        this.onCollision(toCompare)
+                        toCompare.onCollision(this)
+                }
+            }
+        })
```

Comment:
How about we move this ``collisionCheck`` into a ``Geometry`` or ``Utility`` file, so we can reuse the same AABB logic for both Junkerships and Trashbots?

```js
function collisionCheck(a, b) {
    
    var x1 = a.position.x
    var y1 = a.position.y
    var w1 = a.width
    var h1 = a.height
    
    var x2 = b.position.x
    var y2 = b.position.y
    var w2 = b.width
    var h2 = b.height
    
    if(x1 < x2 + w2 && x1 + w1 > x2
    && y1 < y2 + h2 && h1 + y1 > y2) {
        return true
    }
}
```

```js
if(child instanceof Trashbot
|| child instanceof Junkership) {
    if(checkCollision(a)
 +        }
if(child instanceof Trashbot && toCompare instanceof Projectile){